pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -70,8 +71,31 @@ func getAndPublishMetrics(ctx *domainContext, hyper hypervisor.Hypervisor) {
 			// We clear the memory so it doesn't accidentally get
 			// reported.  We keep the CPUTotal and AvailableMemory
 			dm.UsedMemory = 0
+			dm.MaxUsedMemory = 0
 			dm.UsedMemoryPercent = 0
 		}
+
+		config := lookupDomainConfig(ctx, status.Key())
+		if config == nil {
+			continue
+		}
+
+		var CurrMaxUsedMemory uint32
+		st, _ := ctx.pubDomainMetric.Get(dm.Key())
+		if st != nil {
+			previous_metric := st.(types.DomainMetric)
+			CurrMaxUsedMemory = previous_metric.MaxUsedMemory
+		}
+
+		if CurrMaxUsedMemory < dm.MaxUsedMemory {
+			used_pct := dm.MaxUsedMemory * 100 * 1024 / uint32(config.Memory)
+			logrus.Infof("Memory watermark for %s increased: %d MiB,"+
+				" app-memory %d MiB (%d%%), %.2f%% of cgroup limit",
+				status.DomainName,
```

Comment:
status can be nil here. Use config.DomainName instead

Reply:
well, if status is nil, then config is also nil. because
```
config := lookupDomainConfig(ctx, status.Key())
```

As I understand, if `status` is `nil`, the domain is dying, or perhaps has not started yet. So I'll just do nothing it if is `nil`

Reply:
The tests hit a golsng nil pointer panic. Might be due to that. The pr slack channel has a pointer to the Jenkins logs. Don’t know if eden tests failed for the same reason.